app/controllers/application_controller.rb

```diff
@@ -2,10 +2,8 @@ class ApplicationController < ActionController::Base
   include DfE::Analytics::Requests
 
   default_form_builder(GOVUKDesignSystemFormBuilder::FormBuilder)
-
-  http_basic_authenticate_with name: ENV.fetch("SUPPORT_USERNAME", "test"),
-                               password: ENV.fetch("SUPPORT_PASSWORD", "test"),
-                               unless: -> { FeatureFlag.active?(:service_open) }
+  before_action :authenticate_staff!,
```

Comment:
We should be aware that this does mean that users who authenticate using the "test" credentials _will_ have access to the support console while the feature flag is enabled, since they'll be authenticated as staff users. I think this is fine if we're happy for it to work this way, and we're aware of the need to turn off the flag while testing is not going on.

Also worth being aware (and perhaps documenting) that for this to work it will also require `staff_http_basic_auth` to be enabled, as it's only when that feature flag is enabled that the staff HTTP basic auth strategy is used (otherwise we use Devise ordinary username/password strategy against instances of the `Staff` model).

Reply:
The test_user will have access to support if the flag is enabled and the open_service flag is disabled. This is explained in the description of the test_user flag and while not ideal, I don't think will cause any problems as long as we are aware.